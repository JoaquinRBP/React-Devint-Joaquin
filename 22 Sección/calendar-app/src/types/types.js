export const types = {
    uiOpenModal: '[UI] Open Modal',
    uiCloseModal: '[UI] Close Modal',
    eventAddNew: '[event] Add New',
    eventSetActive: '[event] Set Active',
    eventClearActive: '[event] Clear Active',
    eventUpdated: '[event] Update',
    eventDeleted: '[event] Delete',
}