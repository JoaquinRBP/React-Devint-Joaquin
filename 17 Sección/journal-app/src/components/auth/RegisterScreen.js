import React from 'react'
import { Link } from 'react-router-dom'

export const RegisterScreen = () => {
return (
    <>
        <h3 className="auth__title">LoginScreen</h3>
        <form>
            <input 
                type="text"
                placeholder="Name"
                name="name"
                className="auth__input"
            />
            <input 
                type="text"
                placeholder="Email"
                name="email"
                className="auth__input"
            />
            <input 
                type="pasword"
                placeholder="Password"
                name="password"
                className="auth__input"
                autoComplete="off"
            />
            <input 
                type="pasword"
                placeholder="Confirm"
                name="password2"
                className="auth__input"
                autoComplete="off"
            />
            <button 
                type="submit"
                className="btn btn-primary btn-block mb-5"
            >
                Register
            </button>
            <Link className="link mt-5" to="/auth/login">
                All Ready Register?
            </Link>
        </form>
    </>
)
}
