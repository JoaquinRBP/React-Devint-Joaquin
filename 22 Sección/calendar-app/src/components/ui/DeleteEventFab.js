import React from 'react'
import { useDispatch } from 'react-redux'
import { eventClearActiveEvent, eventDeleted } from '../../actions/events';

export const DeleteEventFab = () => {
    const dispatch = useDispatch();
    const handleDeleteEvent=()=>{
        dispatch(eventDeleted());
        dispatch(eventClearActiveEvent())
    }
return (
    <div>
        <button 
        className="btn btn-danger fab-danger"
        onClick={handleDeleteEvent}>
            <i className="fas fa-trash"></i>
            Borrar Evento
        </button>
    </div>
)
}
