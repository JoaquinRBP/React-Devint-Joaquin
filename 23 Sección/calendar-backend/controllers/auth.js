const {response} = require('express');
const { validationResult} = require('express-validator');
const crearUsuario = (req,res = response) => {
    
    const {nombre, email, password} = req.body;
    // if(nombre.length <5) {
    //     return res.status(400).json({
    //         ok: false,
    //         msg:'El nombre debe de ser mayor o igual a 5 letras'
    //     })
    // }

    res.status(201).json(
        {
            ok: true,
            msg:'register',
            nombre,
            email,
            password
        }
        ); 
}
const loginUsuario = (req,res=response) => {
    const {email, password} = req.body
    
    
    res.json({
        ok: true,
        msg:'Login',
        email,
        password
    });
}
const revalidarToken = (req,res=response) => {
    res.json({
        ok: true,
        msg: 'renew'
    });
}
module.exports = {
    crearUsuario,
    loginUsuario,
    revalidadToken: revalidarToken,
};