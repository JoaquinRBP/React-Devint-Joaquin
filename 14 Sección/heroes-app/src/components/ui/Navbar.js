import React, { useContext } from 'react';
import { Link, NavLink, useNavigate } from 'react-router-dom';
import {AuthContext} from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {
    const {user: {name},dispatch} = useContext(AuthContext);
    const navigate = useNavigate();
    const handleLogout = ()=>{
        dispatch({
            type: types.logout,
        })
        navigate('/login',{replace: true});
    }
return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <Link to='/' className='navbar-brand'>14 Secci√≥n</Link>
        <div className='collapse navbar-collapse' id='navbarNavAltMarkup'>
            <div className='navbar-nav'>
                <NavLink exact="true" activeclassname='active' to="./" className='nav-item nav-link'>Home</NavLink>
                <NavLink exact="true" activeclassname='active' to="./marvel" className='nav-item nav-link'>Marvel</NavLink>
                <NavLink exact="true" activeclassname='active' to="./dc" className='nav-item nav-link'>Dc</NavLink>
                <NavLink exact="true" activeclassname='active' to="./search" className='nav-item nav-link'>Search</NavLink>
            </div>
        </div>
        <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
            <ul className="navbar-nav ml-auto">
                <p className="nav-item nav-link text-info">
                    {name}
                </p>
                <button className='nav-item nav-link btn' onClick={handleLogout}>Logout</button>
            </ul>
        </div>
    </nav>
)
}