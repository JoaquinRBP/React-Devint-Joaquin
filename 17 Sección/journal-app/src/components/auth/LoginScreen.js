import React from 'react'
import { Link } from 'react-router-dom'
import {useForm} from '../../hooks/useForm';
import {useDispatch} from 'react-redux';
import validator from 'validator';
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import { EvalError, removeError } from '../../actions/ui';
import { useSelector } from 'react-redux';
export const LoginScreen = () => {
    //useDispatch -- hook
    const dispatch =useDispatch();
    const [values, handleInputChange] = useForm({
        email:'joaquin@gmail.com',
        password:123456
    });
    const {loading} = useSelector( state => state.ui);
    const {email, password} = values;
    const handleLogin = (e) => {
        e.preventDefault();
        if(isFormValid()){
            dispatch(startLoginEmailPassword(email,password));
        }
    }
    const handleGoogleLogin = (e) => {
        e.preventDefault();
        dispatch(startGoogleLogin());
    }
    const isFormValid = () => {
        if(!validator.isEmail(email)){
            dispatch(EvalError('Email is not valid'));
            //console.log('');
            return false;
        }else if(password.length<5){
            dispatch(EvalError('Password is not valid'));
            //console.log('Password is not valid');
            return false;
        }
            dispatch(removeError());
        return true;
    }
return (
    <>
        <h3 className="auth__title">LoginScreen</h3>
        <form 
        className="animate__animated animate__fadeIn animate__faster"
        onSubmit={handleLogin}>
            <input 
                type="text"
                placeholder="Email"
                name="email"
                className="auth__input"
                value={email}
                onChange={handleInputChange}
            />
            <input 
                type="pasword"
                placeholder="Password"
                name="password"
                className="auth__input"
                autoComplete="off"
                value={password}
                onChange={handleInputChange}
            />
            <button 
                type="submit"
                className="btn btn-primary btn-block"
                disabled={loading}
            >
                Login
            </button>
            <div className="auth__social-networks">
                <p>Login with Social Networks</p>
                    <div 
                        className="google-btn"
                        onClick={handleGoogleLogin}
                    >
                    <div className="google-icon-wrapper">
                        <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                    </div>
                    <p className="btn-text">
                        <b>Sign in with google</b>
                    </p>
                    </div>
            </div>
            <Link className="link" to="/auth/register">
                Create a new account
            </Link>
        </form>
    </>
)
}
