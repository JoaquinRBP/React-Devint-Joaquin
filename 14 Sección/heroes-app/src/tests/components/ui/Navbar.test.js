import {shallow, mount} from 'enzyme';
import {Navbar} from '../../../components/ui/Navbar';
import {AuthContext} from '../../../auth/AuthContext';
import { MemoryRouter } from 'react-router';
import { types } from '../../../types/types';
jest.mock('react-router',() => ({
    ...jest.requireActual('react-router'),
    useNavigate: ()=>jest.fn()
}));
const contextValue ={
    dispatch: jest.fn(),
    user:{
        logged:true,
        name: 'Joaquin'
    }
}
describe('Pruebas en Navbar', () => {
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Navbar />
            </MemoryRouter>
        </AuthContext.Provider>
        )
    test('Deberia mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Joaquin');
    });
    test('debe de llamar el logout', () => {
        wrapper.find('button').prop('onClick')();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout,
        });
    })
    
})
