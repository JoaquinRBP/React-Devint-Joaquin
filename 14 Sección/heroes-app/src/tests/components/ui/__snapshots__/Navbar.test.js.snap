// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Navbar Deberia mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    14 Secci√≥n
  </a>
  <div
    className="collapse navbar-collapse"
    id="navbarNavAltMarkup"
  >
    <div
      className="navbar-nav"
    >
      <a
        activeclassname="active"
        aria-current="page"
        className="nav-item nav-link active"
        exact="true"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        activeclassname="active"
        className="nav-item nav-link"
        exact="true"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        activeclassname="active"
        className="nav-item nav-link"
        exact="true"
        href="/dc"
        onClick={[Function]}
      >
        Dc
      </a>
      <a
        activeclassname="active"
        className="nav-item nav-link"
        exact="true"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <p
        className="nav-item nav-link text-info"
      >
        Joaquin
      </p>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
